pyvrp/GeneticAlgorithm.py

```diff
@@ -73,20 +75,27 @@ def __init__(
         population: Population,
         local_search: LocalSearch,
         crossover_op: CrossoverOperator,
+        initial_solutions: Collection[Individual],
         params: GeneticAlgorithmParams = GeneticAlgorithmParams(),
     ):
-        if len(population) == 0:
-            raise ValueError("Expected non-empty population.")
+        if len(initial_solutions) == 0:
+            raise ValueError("Expected at least one initial solution.")
 
         self._data = data
         self._pm = penalty_manager
         self._rng = rng
         self._pop = population
         self._ls = local_search
         self._op = crossover_op
+        self._initial_solutions = initial_solutions
         self._params = params
 
-        self._best = Individual.make_random(data, penalty_manager, rng)
+        # Initialise best using the feasible initial solution with lowest cost,
+        # or the initial solution with lowest cost if none are feasible.
+        self._best = min(
+            self._initial_solutions,
+            key=lambda indiv: (not indiv.is_feasible(), indiv.cost()),
+        )
```

Comment:
Here, and in a test: this is a bit too clever IMHO. I had to think for too long about why this works, so its readability could be improved.

Something like
```python
# Find best feasible initial solution if any exist, else set lower cost
# solution as initial best.
init_feas = [indiv for indiv in initial_solutions if indiv.is_feasible()]
sols = init_feas if init_feas else initial_solutions
self._best = min(sols, key=lambda indiv: indiv.cost())
```
is barely longer, but doesn't make me think as hard.

In the test I think we can add some feasible individual that we know is the best, and then test whether that individual is indeed set as the best. There's no need to reproduce the logic there.